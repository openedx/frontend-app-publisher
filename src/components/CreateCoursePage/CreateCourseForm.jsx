import moment from 'moment';
import React from 'react';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ReduxFormCreatableSelect from '../ReduxFormCreatableSelect';
import RenderInputTextField from '../RenderInputTextField';
import RenderSelectField from '../RenderSelectField';
import ActionButton from '../ActionButton';
import ButtonToolbar from '../ButtonToolbar';
import FieldLabel from '../FieldLabel';
import PriceList from '../PriceList';

import {
  DATE_INPUT_PATTERN,
} from '../../data/constants';
import {
  endDateHelp, runTypeHelp, pacingHelp, startDateHelp, titleHelp, typeHelp, keyHelp,
} from '../../helpText';
import DateTimeField from '../DateTimeField';
import {
  isSafari, localTimeZone, getDateWithDashes, getOptionsData, parseCourseTypeOptions, parseOptions,
} from '../../utils';
import { basicValidate, handleStafferOrCreateFormFail } from '../../utils/validation';

import './CreateCourseForm.scss';

class BaseCreateCourseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { canSetRunKey: false };
  }

  componentDidUpdate(prevProps) {
    const {
      change,
      courseOptions,
      currentFormValues: {
        type: currentType,
        org: currentOrg,
      },
      organizations,
    } = this.props;
    const {
      currentFormValues: {
        type: prevType,
        org: previousOrg,
      },
    } = prevProps;
    if (currentType !== prevType) {
      const courseOptionsData = getOptionsData(courseOptions);
      const parsedTypeOptions = courseOptionsData
        && parseCourseTypeOptions(courseOptionsData.type.type_options);
      const { courseRunTypeOptions } = parsedTypeOptions;
      change('run_type', courseRunTypeOptions[currentType][1].value);
    }
    if (currentOrg !== previousOrg) {
      const selectedOrganization = organizations.find(org => org.key === currentOrg);
      if (selectedOrganization) {
        this.setRunKeyState(selectedOrganization);
      }
    }
  }

  setRunKeyState(selectedOrganization) {
    this.setState({ canSetRunKey: !selectedOrganization.auto_generate_course_run_keys });
  }

  processOrganizations(organizations) {
    let orgSelectList = [];
    if (organizations) {
      const newOrgs = organizations.map(org => (
        { label: org.name, value: org.key, autoGenerateKey: org.auto_generate_course_run_keys }
      ));
      newOrgs.sort((a, b) => a.label.localeCompare(b.label, undefined, { sensitivity: 'base' }));
      orgSelectList = orgSelectList.concat(newOrgs);
    }
    return orgSelectList;
  }

  // TODO: Removing this for now, as we hide the source field for now and it's not being used.
  //   processSources(sources) {
  //     let sourceSelectList = [{ label: 'Select product source', value: '' }];
  //     if (sources) {
  //       const newSources = sources.map(source => (
  //         { label: source.name, value: source.slug }
  //       ));
  //       sourceSelectList = sourceSelectList.concat(newSources);
  //     }
  //     return sourceSelectList;
  //   }

  render() {
    const {
      currentFormValues,
      handleSubmit,
      organizations,
      pristine,
      isCreating,
      courseOptions,
      courseRunOptions,
    } = this.props;
    const { canSetRunKey } = this.state;
    const courseOptionsData = getOptionsData(courseOptions);
    const parsedTypeOptions = courseOptionsData
      && parseCourseTypeOptions(courseOptionsData.type.type_options);
    const { courseTypeOptions } = parsedTypeOptions;
    const { courseRunTypeOptions } = parsedTypeOptions;
    const { priceLabels } = parsedTypeOptions;
    const courseRunOptionsData = getOptionsData(courseRunOptions);
    const { pacing_type: { choices } } = courseRunOptionsData;
    const pacingTypeOptions = courseRunOptionsData && parseOptions(choices);

    return (
      <div className="create-course-form">
        <h2>Create New Course</h2>
        <hr />
        <form onSubmit={handleSubmit}>
          <Field
            name="org"
            component={ReduxFormCreatableSelect}
            label={<FieldLabel text="Organization" />}
            options={this.processOrganizations(organizations)}
            validate={basicValidate}
            placeholder="Organization..."
          />
          {/* TODO: Removing this field because publisher is currently only used for creating Edx courses.
            We will unhide this field once we start supporting other products as well in publisher. */}
          {/* <Field
            name="source"
            component={RenderSelectField}
            options={this.processSources(sources)}
            label={<FieldLabel text="Source" required />}
            defaultValue={DEFAULT_PRODUCT_SOURCE}
            hidden
            required
          /> */}
          <Field
            name="title"
            component={RenderInputTextField}
            type="text"
            label={(
              <FieldLabel
                id="title-label"
                text="Title"
                required
                helpText={titleHelp}
              />
            )}
            required
          />
          <Field
            name="number"
            component={RenderInputTextField}
            type="text"
            label={(
              <FieldLabel
                id="number-label"
                text="Number"
                required
                extraText="Cannot edit after submission"
                helpText={(
                  <div>
                    <p>
                      Maximum 10 characters. Characters can be letters, numbers, periods,
                      underscores, or hyphens.
                    </p>
                    <p>
                      If a course consists of several modules, the course number can have an
                      ending such as .1x or .2x.
                    </p>
                    <p>
                      <a
                        href="https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/set_up_course/planning_course_information/title_number_guidelines.html#course-number-guidelines"
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        Learn more.
                      </a>
                    </p>
                    <p><b>Examples:</b></p>
                    <ol>
                      <li>CS002x</li>
                      <li>BIO1.1x, BIO1.2x</li>
                    </ol>
                  </div>
                )}
              />
            )}
            required
          />
          <Field
            name="type"
            component={RenderSelectField}
            options={courseTypeOptions}
            label={(
              <FieldLabel
                id="course-type-label"
                text="Course enrollment track"
                required
                helpText={typeHelp}
              />
            )}
            required
          />
          <PriceList
            priceLabels={currentFormValues.type ? priceLabels[currentFormValues.type] : {}}
            required
          />
          <h2>First run of your Course</h2>
          <hr />
          {canSetRunKey
            && (
            <Field
              name="courseRunKey"
              component={RenderInputTextField}
              type="text"
              pattern="[a-zA-Z0-9-]+"
              label={(
                <FieldLabel
                  id="courseRunKey-label"
                  text="Run Key"
                  helpText={keyHelp}
                  optional
                />
              )}
            />
            )}
          {/* TODO this should be refactored when paragon supports safari */}
          {/* text inputs for safari */}
          {isSafari
            ? (
              <div>
                <Field
                  name="start"
                  type="text"
                  component={DateTimeField}
                  dateLabel="Start date"
                  timeLabel={`Start time (${localTimeZone})`}
                  helpText={startDateHelp}
                  required
                  maxLength="10"
                  pattern={DATE_INPUT_PATTERN}
                  placeholder="yyyy/mm/dd"
                />
                <Field
                  name="end"
                  type="text"
                  component={DateTimeField}
                  dateLabel="End date"
                  timeLabel={`End time (${localTimeZone})`}
                  helpText={endDateHelp}
                  required
                  maxLength="10"
                  pattern={DATE_INPUT_PATTERN}
                  placeholder="yyyy/mm/dd"
                />
              </div>
            )
            // date inputs for all browsers besides safari
            : (
              <div>
                <Field
                  name="start"
                  type="date"
                  component={DateTimeField}
                  dateLabel="Start date"
                  timeLabel={`Start time (${localTimeZone})`}
                  helpText={startDateHelp}
                  required
                  minDate={getDateWithDashes(moment())}
                />
                <Field
                  name="end"
                  type="date"
                  component={DateTimeField}
                  dateLabel="End date"
                  timeLabel={`End time (${localTimeZone})`}
                  helpText={endDateHelp}
                  required
                  minDate={getDateWithDashes(moment(currentFormValues.start).add(1, 'd') || moment())}
                />
              </div>
            )}
          <Field
            name="run_type"
            component={RenderSelectField}
            options={currentFormValues.type ? courseRunTypeOptions[currentFormValues.type] : [{ label: 'Select Course enrollment track first', value: '' }]}
            label={(
              <FieldLabel
                id="course-run-type-label"
                text="Course run enrollment track"
                required
                helpText={runTypeHelp}
              />
            )}
            required
          />
          <Field
            name="pacing_type"
            type="text"
            component={RenderSelectField}
            options={pacingTypeOptions}
            label={(
              <FieldLabel
                id="pacing_type.label"
                text="Course pacing"
                helpText={pacingHelp}
              />
            )}
          />
          <ButtonToolbar>
            <Link to="/">
              <button
                type="button"
                className="btn btn-outline-primary"
                disabled={isCreating}
              >
                Cancel
              </button>
            </Link>
            <ActionButton
              disabled={pristine}
              labels={{
                default: 'Create',
                pending: 'Creating',
              }}
              state={isCreating ? 'pending' : 'default'}
            />
          </ButtonToolbar>
        </form>
      </div>
    );
  }
}

BaseCreateCourseForm.defaultProps = {
  change: () => null,
  isCreating: false,
  pristine: true,
  currentFormValues: {},
};

BaseCreateCourseForm.propTypes = {
  change: PropTypes.func,
  handleSubmit: PropTypes.func.isRequired,
  initialValues: PropTypes.shape({ // eslint-disable-line react/no-unused-prop-types
    org: PropTypes.string,
    source: PropTypes.string,
    title: PropTypes.string,
    number: PropTypes.string,
    type: PropTypes.string,
    prices: PropTypes.shape(),
    start: PropTypes.string,
    end: PropTypes.string,
    upgrade_deadline_override: PropTypes.string,
  }).isRequired,
  currentFormValues: PropTypes.shape({
    org: PropTypes.string,
    type: PropTypes.string,
    start: PropTypes.string,
  }),
  organizations: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    key: PropTypes.string.isRequired,
  })).isRequired,
  pristine: PropTypes.bool,
  isCreating: PropTypes.bool,
  courseOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }).isRequired,
  courseRunOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }).isRequired,
};

export default reduxForm({
  form: 'create-course-form',
  onSubmitFail: handleStafferOrCreateFormFail,
})(BaseCreateCourseForm);
export { BaseCreateCourseForm };
