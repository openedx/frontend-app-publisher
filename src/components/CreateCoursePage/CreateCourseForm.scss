.create-course-form {
    // react-select specific styles
    .react-select__control {
      cursor: text;
      border-color: #707070;
      // match the height of edx-paragon form fields
      height: calc(1.3333em + 1.125rem + 2px);
    }
    
    .react-select__dropdown-indicator{
        color: #707070;
    }
  
    .react-select__control--menu-is-open{
      .react-select__single-value{
        color: #D3D3D3
      }
    }
  }
