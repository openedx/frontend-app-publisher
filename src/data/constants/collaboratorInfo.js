export const CREATE_COLLABORATOR = 'CREATE_COLLABORATOR';
export const CREATE_COLLABORATOR_SUCCESS = 'CREATE_COLLABORATOR_SUCCESS';
export const CREATE_COLLABORATOR_FAIL = 'CREATE_COLLABORATOR_FAIL';
export const EDIT_COLLABORATOR_INFO = 'EDIT_COLLABORATOR_INFO';
export const EDIT_COLLABORATOR_INFO_SUCCESS = 'EDIT_COLLABORATOR_INFO_SUCCESS';
export const EDIT_COLLABORATOR_INFO_FAIL = 'EDIT_COLLABORATOR_INFO_FAIL';
export const CANCEL_COLLABORATOR_INFO = 'CANCEL_COLLABORATOR_INFO';
export const RESET_COLLABORATOR_INFO = 'RESET_COLLABORATOR_INFO';
