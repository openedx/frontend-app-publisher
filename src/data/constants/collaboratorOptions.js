export const REQUEST_COLLABORATOR_OPTIONS = 'REQUEST_COLLABORATOR_OPTIONS';
export const REQUEST_COLLABORATOR_OPTIONS_FAIL = 'REQUEST_COLLABORATOR_OPTIONS_FAIL';
export const REQUEST_COLLABORATOR_OPTIONS_SUCCESS = 'REQUEST_COLLABORATOR_OPTIONS_SUCCESS';
