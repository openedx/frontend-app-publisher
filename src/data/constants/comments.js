const CREATE_COMMENT = 'CREATE_COMMENT';
const CREATE_COMMENT_FAIL = 'CREATE_COMMENT_FAIL';
const CREATE_COMMENT_SUCCESS = 'CREATE_COMMENT_SUCCESS';
const REQUEST_COMMENTS = 'REQUEST_COMMENTS';
const REQUEST_COMMENTS_FAIL = 'REQUEST_COMMENTS_FAIL';
const REQUEST_COMMENTS_SUCCESS = 'REQUEST_COMMENTS_SUCCESS';

export {
  CREATE_COMMENT,
  CREATE_COMMENT_FAIL,
  CREATE_COMMENT_SUCCESS,
  REQUEST_COMMENTS,
  REQUEST_COMMENTS_FAIL,
  REQUEST_COMMENTS_SUCCESS,
};
