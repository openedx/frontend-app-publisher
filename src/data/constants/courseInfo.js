const REQUEST_COURSE_INFO_FAIL = 'REQUEST_COURSE_INFO_FAIL';
const REQUEST_COURSE_INFO_SUCCESS = 'REQUEST_COURSE_INFO_SUCCESS';
const REQUEST_COURSE_INFO = 'REQUEST_COURSE_INFO';
const CREATE_COURSE = 'CREATE_COURSE';
const CREATE_COURSE_SUCCESS = 'CREATE_COURSE_SUCCESS';
const CREATE_COURSE_FAIL = 'CREATE_COURSE_FAIL';
const EDIT_COURSE_INFO = 'EDIT_COURSE_INFO';
const EDIT_COURSE_SUCCESS = 'EDIT_COURSE_SUCCESS';
const EDIT_COURSE_FAIL = 'EDIT_COURSE_FAIL';
const CREATE_COURSE_RUN = 'CREATE_COURSE_RUN';
const CREATE_COURSE_RUN_SUCCESS = 'CREATE_COURSE_RUN_SUCCESS';
const CREATE_COURSE_RUN_FAIL = 'CREATE_COURSE_RUN_FAIL';

const UUID_REGEX = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;

export {
  REQUEST_COURSE_INFO_FAIL,
  REQUEST_COURSE_INFO_SUCCESS,
  REQUEST_COURSE_INFO,
  CREATE_COURSE,
  CREATE_COURSE_SUCCESS,
  CREATE_COURSE_FAIL,
  EDIT_COURSE_INFO,
  EDIT_COURSE_SUCCESS,
  EDIT_COURSE_FAIL,
  CREATE_COURSE_RUN,
  CREATE_COURSE_RUN_SUCCESS,
  CREATE_COURSE_RUN_FAIL,
  UUID_REGEX,
};
