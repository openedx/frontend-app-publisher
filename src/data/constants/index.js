const VERIFIED_TRACK = Object.freeze({ name: 'Verified and Audit', key: 'verified' });
const AUDIT_TRACK = Object.freeze({ name: 'Audit Only', key: 'audit' });
const PROFESSIONAL_TRACK = Object.freeze({ name: 'Professional Only', key: 'professional' });
const MASTERS_TRACK = Object.freeze({ key: 'masters' });
const ENTITLEMENT_TRACKS = Object.freeze(['verified', 'professional']);

const REVIEW_BY_LEGAL = 'review_by_legal';
const REVIEW_BY_INTERNAL = 'review_by_internal';
const PUBLISHED = 'published';
const REVIEWED = 'reviewed';
const UNPUBLISHED = 'unpublished';
const ARCHIVED = 'archived';
const IN_REVIEW_STATUS = Object.freeze([REVIEW_BY_LEGAL, REVIEW_BY_INTERNAL]);
const POST_REVIEW_STATUSES = Object.freeze([PUBLISHED, REVIEWED]);
const COURSE_EXEMPT_FIELDS = Object.freeze(['imgSrc', 'videoSrc']);
const COURSE_RUN_NON_EXEMPT_FIELDS = Object.freeze(['expected_program_name', 'expected_program_type']);

const DATE_FORMAT = 'YYYY-MM-DD[T]HH:mm:ss[Z]';
const DATE_INPUT_PATTERN = '20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])';
const FORMAT_DATE_MATCHER = /20\d{2}-(0\d{1}|1[0-2])-([0-2]\d{1}|3[0-1])/;
const NORMALIZE_DATE_MATCHER = /20\d{2}\/(0\d{1}|1[0-2])\/([0-2]\d{1}|3[0-1])/;

const EXECUTIVE_EDUCATION_SLUG = 'executive-education-2u';
const BOOTCAMP_SLUG = 'bootcamp-2u';

const COURSE_URL_SLUG_PATTERN_OLD = '^[a-z0-9_]+(?:-[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN_NEW = '^learn/[a-z0-9_]+(?:-?[a-z0-9_]+)*/[a-z0-9_]+(?:-?[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN = `${COURSE_URL_SLUG_PATTERN_NEW}|${COURSE_URL_SLUG_PATTERN_OLD}`;

const COURSE_URL_SLUG_VALIDATION_MESSAGE = {
  [COURSE_URL_SLUG_PATTERN_OLD]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only.',
  [COURSE_URL_SLUG_PATTERN_NEW]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format learn/<primary_subject>/<org-slug>-<course_slug>',
  [COURSE_URL_SLUG_PATTERN]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format <custom-url-slug> or learn/<primary_subject>/<org-slug>-<course_slug>.',
};

export {
  VERIFIED_TRACK,
  AUDIT_TRACK,
  PROFESSIONAL_TRACK,
  MASTERS_TRACK,
  ENTITLEMENT_TRACKS,
  REVIEW_BY_LEGAL,
  REVIEW_BY_INTERNAL,
  PUBLISHED,
  REVIEWED,
  UNPUBLISHED,
  ARCHIVED,
  IN_REVIEW_STATUS,
  POST_REVIEW_STATUSES,
  DATE_FORMAT,
  DATE_INPUT_PATTERN,
  FORMAT_DATE_MATCHER,
  NORMALIZE_DATE_MATCHER,
  COURSE_EXEMPT_FIELDS,
  COURSE_RUN_NON_EXEMPT_FIELDS,
  EXECUTIVE_EDUCATION_SLUG,
  BOOTCAMP_SLUG,
  COURSE_URL_SLUG_PATTERN,
  COURSE_URL_SLUG_PATTERN_NEW,
  COURSE_URL_SLUG_PATTERN_OLD,
  COURSE_URL_SLUG_VALIDATION_MESSAGE,
};
