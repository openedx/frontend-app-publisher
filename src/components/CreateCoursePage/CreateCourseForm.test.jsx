import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { BaseCreateCourseForm } from './CreateCourseForm';
import { courseOptions, courseRunOptions } from '../../data/constants/testData';

const organizations = [
  { name: 'edX', key: 'edx' }, { name: 'edX2', key: 'edx2' },
];

const sources = [
  { name: 'souce1', slug: 'source1' }, { name: 'source2', slug: 'source2' },
];

Date.now = jest.fn(() => new Date(Date.UTC(2001, 0, 1)).valueOf());

describe('CreateCourseForm', () => {
  const initialValues = {
    org: 'edx',
    title: 'Hello',
    number: 'edx101',
    type: '8a8f30e1-23ce-4ed3-a361-1325c656b67b',
    prices: {
      verified: '100.00',
    },
  };

  it('renders html correctly with no orgs', () => {
    const component = shallow(<BaseCreateCourseForm
      handleSubmit={() => {}}
      initialValues={{}}
      organizations={organizations}
      sources={sources}
      courseOptions={courseOptions}
      courseRunOptions={courseRunOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders html correctly with no sources', () => {
    const component = shallow(<BaseCreateCourseForm
      handleSubmit={() => {}}
      initialValues={{}}
      organizations={organizations}
      sources={[]}
      courseOptions={courseOptions}
      courseRunOptions={courseRunOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders html correctly with data', () => {
    const component = shallow(<BaseCreateCourseForm
      handleSubmit={() => {}}
      initialValues={initialValues}
      currentFormValues={initialValues}
      organizations={organizations}
      sources={sources}
      courseOptions={courseOptions}
      courseRunOptions={courseRunOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders html correctly while submitting', () => {
    const component = shallow(<BaseCreateCourseForm
      submitting
      isCreating
      pristine={false}
      handleSubmit={() => {}}
      initialValues={{}}
      currentFormValues={initialValues}
      organizations={organizations}
      sources={sources}
      courseOptions={courseOptions}
      courseRunOptions={courseRunOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
