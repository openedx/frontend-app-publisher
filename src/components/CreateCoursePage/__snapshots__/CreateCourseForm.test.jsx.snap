// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCourseForm renders html correctly while submitting 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    handleSubmit={[Function]}
    initialValues={
      Object {
        "enrollmentTrack": "",
        "number": "",
        "org": "",
        "price": "",
        "title": "",
      }
    }
    organizations={
      Array [
        Object {
          "key": "edx",
          "name": "edX",
        },
        Object {
          "key": "edx2",
          "name": "edX2",
        },
      ]
    }
    submitting={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "create-course-form",
      "getFormState": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "enrollmentTrack": "",
        "number": "",
        "org": "",
        "price": "",
        "title": "",
      },
      "keepDirtyOnReinitialize": false,
      "organizations": Array [
        Object {
          "key": "edx",
          "name": "edX",
        },
        Object {
          "key": "edx2",
          "name": "edX2",
        },
      ],
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "submitting": true,
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "create-course-form",
        "getFormState": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "enrollmentTrack": "",
          "number": "",
          "org": "",
          "price": "",
          "title": "",
        },
        "keepDirtyOnReinitialize": false,
        "organizations": Array [
          Object {
            "key": "edx",
            "name": "edX",
          },
          Object {
            "key": "edx2",
            "name": "edX2",
          },
        ],
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "submitting": true,
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`CreateCourseForm renders html correctly with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    handleSubmit={[Function]}
    initialValues={
      Object {
        "enrollmentTrack": "verified",
        "number": "edx101",
        "org": "edx",
        "price": 100,
        "title": "Hello",
      }
    }
    organizations={
      Array [
        Object {
          "key": "edx",
          "name": "edX",
        },
        Object {
          "key": "edx2",
          "name": "edX2",
        },
      ]
    }
    pristine={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "create-course-form",
      "getFormState": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "enrollmentTrack": "verified",
        "number": "edx101",
        "org": "edx",
        "price": 100,
        "title": "Hello",
      },
      "keepDirtyOnReinitialize": false,
      "organizations": Array [
        Object {
          "key": "edx",
          "name": "edX",
        },
        Object {
          "key": "edx2",
          "name": "edX2",
        },
      ],
      "persistentSubmitErrors": false,
      "pristine": true,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "create-course-form",
        "getFormState": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "enrollmentTrack": "verified",
          "number": "edx101",
          "org": "edx",
          "price": 100,
          "title": "Hello",
        },
        "keepDirtyOnReinitialize": false,
        "organizations": Array [
          Object {
            "key": "edx",
            "name": "edX",
          },
          Object {
            "key": "edx2",
            "name": "edX2",
          },
        ],
        "persistentSubmitErrors": false,
        "pristine": true,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`CreateCourseForm renders html correctly with no orgs 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "create-course-form",
      "getFormState": [Function],
      "initialValues": undefined,
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "create-course-form",
        "getFormState": [Function],
        "initialValues": undefined,
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
